code/game/machinery/computer/medical.dm

```diff
@@ -55,19 +52,66 @@
 <BR><A href='?src=[REF(src)];logout=1'>{Log Out}</A><BR>
 "}
 				if(2)
+					// JS scripts for the filter
+					dat += {"
+					<script src="[SSassets.transport.get_asset_url("jquery.min.js")]"></script>
+					<script type='text/javascript'>
+
+						function updateSearch(){
+							var filter_text = document.getElementById('filter');
+							var filter = filter_text.value.toLowerCase();
+
+							if(complete_list != null && complete_list != ""){
+								var mtbl = document.getElementById("maintable_data_archive");
+								mtbl.innerHTML = complete_list;
+							}
+
+							if(filter.value == ""){
+								return;
+							}else{
+								$("#maintable_data").children("tbody").children("tr").children("td").children("input").filter(function(index)
```

Comment:
What's weird about it is that there is no `tbody` in the code, but without  the `.children("tbody")` it refuses to work. I assumed that it somehow 'injects' `tbody` into every table in the output HTML (which is probably wrong but I can't come up with any other answer) and that's why it can't just find `tr` children in the `#maintable_data`, but I don't know of any way to inspect the HTML so 🤷

Reply:
Looks like this is one of those fucked up HTML moments

> [Tag omission in text/html](https://html.spec.whatwg.org/multipage/dom.html#concept-element-tag-omission):
    A [tbody](https://html.spec.whatwg.org/multipage/tables.html#the-tbody-element) element's [start tag](https://html.spec.whatwg.org/multipage/syntax.html#syntax-start-tag) can be omitted if the first thing inside the [tbody](https://html.spec.whatwg.org/multipage/tables.html#the-tbody-element) element is a [tr](https://html.spec.whatwg.org/multipage/tables.html#the-tr-element) element, and if the element is not immediately preceded by a [tbody](https://html.spec.whatwg.org/multipage/tables.html#the-tbody-element), [thead](https://html.spec.whatwg.org/multipage/tables.html#the-thead-element), or [tfoot](https://html.spec.whatwg.org/multipage/tables.html#the-tfoot-element) element whose [end tag](https://html.spec.whatwg.org/multipage/syntax.html#syntax-end-tag) has been omitted. (It can't be omitted if the element is empty.)
    A [tbody](https://html.spec.whatwg.org/multipage/tables.html#the-tbody-element) element's [end tag](https://html.spec.whatwg.org/multipage/syntax.html#syntax-end-tag) can be omitted if the [tbody](https://html.spec.whatwg.org/multipage/tables.html#the-tbody-element) element is immediately followed by a [tbody](https://html.spec.whatwg.org/multipage/tables.html#the-tbody-element) or [tfoot](https://html.spec.whatwg.org/multipage/tables.html#the-tfoot-element) element, or if there is no more content in the parent element.

You're allowed to omit both the start and end tags for `tbody`, in which case you're still defining a `tbody` without having to write it at all. I think it'd be a good idea to explicitly add it into the table for clarity.

Reply:
Bruh, the more you know... You're right, for clarity's sake it's better if explicitly added.